static/app/views/alerts/incidentRules/triggers/chart/index.tsx

```diff
@@ -164,43 +200,147 @@ class TriggersChart extends React.PureComponent<Props, State> {
-      this.setState({totalEvents: null});
+      this.setState({totalCount: null});
     }
   }
 
-  render() {
-    const {
-      api,
-      organization,
-      projects,
-      timeWindow,
-      query,
-      aggregate,
-      triggers,
-      resolveThreshold,
-      thresholdType,
-      environment,
-      header,
-    } = this.props;
-    const {statsPeriod, totalEvents} = this.state;
+  async fetchSessionTimeSeries() {
+    const {api, organization, environment, projects, query, timeWindow, aggregate} =
+      this.props;
+    try {
+      this.setState(state => ({
+        sessionsLoading: state.sessionTimeSeries === null,
+        sessionsReloading: state.sessionTimeSeries !== null,
+      }));
+      const {groups, intervals}: SessionApiResponse = await api.requestPromise(
+        `/organizations/${organization.slug}/sessions/`,
+        {
+          query: {
+            project: projects.map(({id}) => id),
+            environment: environment ? [environment] : [],
+            statsPeriod: this.getStatsPeriod(),
+            field: SESSION_AGGREGATE_TO_FIELD[aggregate],
+            interval: TIME_WINDOW_TO_SESSION_INTERVAL[timeWindow],
+            groupBy: ['session.status'],
+            query,
+          },
+        }
+      );
+      const totalCount = getCount(groups, SESSION_AGGREGATE_TO_FIELD[aggregate]);
+      const sessionTimeSeries = [
+        {
+          seriesName:
+            AlertWizardAlertNames[
+              getAlertTypeFromAggregateDataset({aggregate, dataset: Dataset.SESSIONS})
+            ],
+          data: getCrashFreeRateSeries(
+            groups,
+            intervals,
+            SESSION_AGGREGATE_TO_FIELD[aggregate]
+          ),
+        },
+      ];
+      this.setState({
+        sessionTimeSeries,
+        totalCount,
+        sessionsLoading: false,
+        sessionsReloading: false,
+      });
+    } catch (e) {
+      this.setState({
+        sessionTimeSeries: null,
+        totalCount: null,
+        sessionsLoading: false,
+        sessionsReloading: false,
+      });
+    }
+  }
 
+  renderChart(
+    data: Series[] = [],
+    isLoading: boolean,
+    isReloading: boolean,
+    maxValue?: number
+  ) {
+    const {triggers, resolveThreshold, thresholdType, header, timeWindow} = this.props;
+    const {statsPeriod, totalCount} = this.state;
     const statsPeriodOptions = AVAILABLE_TIME_PERIODS[timeWindow];
     const period = this.getStatsPeriod();
-
     return (
+      <React.Fragment>
+        {header}
+        <TransparentLoadingMask visible={isReloading} />
+        {isLoading ? (
+          <ChartPlaceholder />
+        ) : (
+          <ThresholdsChart
+            period={statsPeriod}
+            maxValue={maxValue}
+            data={data}
+            triggers={triggers}
+            resolveThreshold={resolveThreshold}
+            thresholdType={thresholdType}
+          />
+        )}
+        <ChartControls>
+          <InlineContainer>
+            <React.Fragment>
```

Comment:
do you need a fragment here?

Reply:
I guess not, just moved that block of code 🤷‍♂️
Will try to remove it.